src/safeposix/syscalls/net_calls.rs

```diff
@@ -74,13 +74,15 @@ impl Cage {
         }
         0
     }
-
+//Creates a new socket, ensuring the requested domain, socket type, and protocol are supported by SafePOSIX and also
+//validates the requested communication domain, socket type, and protocol permitting only combinations that are known
     pub fn socket_syscall(&self, domain: i32, socktype: i32, protocol: i32) -> i32 {
         let real_socktype = socktype & 0x7; //get the type without the extra flags, it's stored in the last 3 bits
-        let nonblocking = (socktype & SOCK_NONBLOCK) != 0;
+        let nonblocking = (socktype & SOCK_NONBLOCK) != 0; // Checks if the socket should be non-blocking.
```

Comment:
Few applications require blocking sockets (where they can wait for an operation to complete) and non-blocking sockets(where they can't afford to wait ).
This is a necessary check; SafePOSIX wouldn't know the application's intended socket behaviour, and the application might fall into a busy-wait loop, resulting in excessive CPU resources.

Reply:
Okay, but why does SafePOSIX need this information?  The kernel will have it because the flags will be passed down.  Why do we also check it?

Reply:
You are right, considering operating system kernels are highly optimized for handling low-level operations like socket management.

Relying solely on the kernel to handle non-blocking behaviour without considering the specific context of the sandboxed environment can undermine the control, security, and efficiency that SafePOSIX is designed to provide.

By performing its own SOCK_NONBLOCK check, SafePOSIX ensures consistent behaviour and maintains its configured operation—this added layer of control and security safeguards against potential issues within its sandboxed environment.

Reply:
ill update this in the comments

Reply:
>By performing its own SOCK_NONBLOCK check, SafePOSIX ensures consistent behaviour and maintains its configured operation—this added layer of control and security safeguards against potential issues within its sandboxed environment.

Okay, be sure to explain how RustPOSIX does this.  (Note, I'm not sure what you're saying here is true.  It may be, but it also may not be.  It's fine to say you don't know why something happens.  As you go through more of the code, you may figure it out and then update your comment...)

Reply:
I wasn't precise before. I've now understood why SafePOSIX requires this and have made the necessary updates.